rpm/gen-fixtures-delta.sh

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env sh
+#
+# Generate RPM repository with DRPM fixture data. 
+# Several RPM utilities must be installed, including `createrepo` and `makedeltarpm`.
+#
+set -euo pipefail
+
+# Print a message to stdout explaining how to use this script.
+#
+# NOTE: $0 corresponds to the the script's name only in some shells and in some
+# contexts.
+show_help() {
+fmt <<EOF
+usage: gen-fixtures-delta.sh <output_dir> <assets_dir> <dpkg_name>
+
+Create <output_dir>. Use packages in <assets_dir> to generate DRPM fixture data 
+for package <dpkg_name> and place the results into <output_dir>. 
+A suggested name for <output_dir> is "zoo".
+
+EOF
+cat <<EOF
+<output_dir>
+    The directory into which generated fixtures are placed.
+<assets_dir>
+    The directory from which source material is read.
+ <dpkg_name>
+    The name of package which delta RPM will be generated for
+EOF
+}
+
+# Given a path to an RPM package, echo the package name.
+#
+# e.g. assets/walrus-baby-5.21-1.noarch.rpm → walrus-baby
+get_rpm_name() {
+    local filename="$(basename "${1}")"
+    IFS=$'\n' local parts=($(echo "${filename}" | tr - "\n"))
+    local end=$(expr ${#parts[@]} - 2)
+    for ((i=0; i<${end}; i++)); do
+        rpm_name+="${parts[i]}"-
+    done
+    rpm_name="${rpm_name::-1}"
+    echo "${rpm_name#*\!}"  # strip epoch
+}
+
+# Given a path to an RPM package, echo the package version.
+#
+# e.g. assets/walrus-5.21-1.noarch.rpm → 5.21
+get_rpm_version() {
+    local filename="$(basename "${1}")"
+    IFS=$'\n' local parts=($(echo "${filename}" | tr - "\n"))
```

Comment:
I *just* barely learned about the value of declaring variables separately from using them. You can see an example of the dangers [here](http://pastebin.com/m9VPQ1KS). In short, write this instead:

```bash
    local filename parts
    filename="$(basename "${1}")"
    IFS=$'\n' parts=($(echo "${filename}" | tr - "\n"))
```